backend/tests/unit/git/test_git_repository.py

```diff
@@ -279,6 +280,17 @@ async def test_get_branches_from_remote(git_repo_01: InfrahubRepository):
     assert sorted(remote_branches.keys()) == ["branch01", "branch02", "clean-branch", "main"]
 
 
+async def test_get_branches_from_remote_in_import_sync_branch_names(git_repo_01: InfrahubRepository):
+    config.SETTINGS.git.import_sync_branch_names = ["branch.*"]
+    repo = git_repo_01
+
+    remote_branches = repo.get_branches_from_remote()
+    assert isinstance(remote_branches, dict)
+    assert sorted(remote_branches.keys()) == ["branch01", "branch02"]
+
+    config.SETTINGS.git.import_sync_branch_names = []
```

Comment:
Instead of changing the config like this the test and resetting it back to another value afterwards I'd suggest that we use a fixture similar to this one:

```python
@pytest.fixture
def query_limit_of_one() -> Generator[None, None, None]:
    original_query_size_limit = config.SETTINGS.database.query_size_limit
    config.SETTINGS.database.query_size_limit = 1
    yield
    config.SETTINGS.database.query_size_limit = original_query_size_limit
```